modules/xfem/src/base/XFEM.C

```diff
@@ -1353,3 +1382,85 @@ XFEM::getXFEMWeights(MooseArray<Real> &weights, const Elem * elem, QBase * qrule
   }
   return have_weights;
 }
+
+void
+XFEM::getXFEMIntersectionInfo(const Elem* elem, unsigned int plane_id, Point & normal, std::vector<Point> & intersectionPoints, bool displaced_mesh) const
+{
+  std::map<unique_id_type, XFEMCutElem*>::const_iterator it;
+  it = _cut_elem_map.find(elem->unique_id());
+  if (it != _cut_elem_map.end())
+  {
+    const XFEMCutElem *xfce = it->second;
+    if (displaced_mesh)
+      xfce->getIntersectionInfo(plane_id, normal, intersectionPoints, _mesh2);
+    else
+      xfce->getIntersectionInfo(plane_id, normal, intersectionPoints);
+  }
+}
+
+void
+XFEM::getXFEMqRuleOnLine(std::vector<Point> & intersection_points, std::vector<Point> & quad_pts, std::vector<Real> & quad_wts) const
+{
+  Point p1 = intersection_points[0];
+  Point p2 = intersection_points[1];
+
+  //number of quadrature points
+  unsigned int num_qpoints = 2;
+
+  //quadrature coordinates
+  Real xi0 = -std::sqrt(1.0/3.0);
+  Real xi1 =  std::sqrt(1.0/3.0);
+
+  quad_wts.resize(num_qpoints);
+  quad_pts.resize(num_qpoints);
+
+  Real integ_jacobian =  pow((p1 -  p2).size_sq(), 0.5) * 0.5;
+
+  quad_wts[0] = 1.0 * integ_jacobian;
+  quad_wts[1] = 1.0 * integ_jacobian;
+
+  quad_pts[0] = (1.0 - xi0) / 2.0 * p1 + (1.0 + xi0) / 2.0 * p2;
+  quad_pts[1] = (1.0 - xi1) / 2.0 * p1 + (1.0 + xi1) / 2.0 * p2;
+}
+
+void
+XFEM::getXFEMqRuleOnSurface(std::vector<Point> & intersection_points, std::vector<Point> & quad_pts, std::vector<Real> & quad_wts) const
+{
+  unsigned nnd_pe = intersection_points.size();
+  Point xcrd(0.0, 0.0, 0.0);
+  for (unsigned int i = 0; i < intersection_points.size(); ++i)
```

Comment:
you just declared nnd_pe (which should be const, just like in the function above). Why don't you use it in the loop?